expan/core/results.py

```diff
@@ -440,9 +445,16 @@ def to_json(self, fpath=None):
 				else:
 					metrics.append({"name": metric, "subgroup_metrics": subgroup_metrics})
 			variants.append({"name": variant, "metrics": metrics})
-
 		json_tree['variants'] = variants
-		json_tree['metadata'] = self.metadata
+
+		# store metadata in temporary variable as UserWarning() needs to be converted to string so that JSON serialization can continue
+		metadata = self.metadata
+		for m in metadata:
+			if m == 'errors' or m == 'warnings':
+				for k in metadata[m]:
+					metadata[m][k] = str(metadata[m][k])
```

Comment:
definitely, will change this in the next release! sorry for rushing merging this pull request but there is a dependency on having this deployed.